Wikipedia/View Controllers/RecentSearches/RecentSearchesViewController.m

```diff
@@ -39,15 +34,16 @@ - (void)viewDidLoad {
     [self setupTable];
 
     [self updateTrashButtonEnabledState];
-
-    [self.table setBackgroundColor:[UIColor clearColor]];
-    self.table.backgroundView.backgroundColor = [UIColor clearColor];
-    self.table.rowHeight                      = cellHeight;
 }
 
 - (void)setupTable {
-    self.table.separatorStyle = UITableViewCellSeparatorStyleNone;
     [self.table registerNib:[UINib nibWithNibName:@"RecentSearchCell" bundle:nil] forCellReuseIdentifier:@"RecentSearchCell"];
+
+    self.table.estimatedRowHeight = 52.f;
+    self.table.rowHeight          = UITableViewAutomaticDimension;
+
+    [self.table setBackgroundView:nil];
```

Comment:
@montehurd thanks for the note, but it's better to put documentation or explanations with the source so it doesn't get lost.  in particular, this kind of comment (which is a workaround for built-in framework behavior) has been given `HAX:` prefix to signal to readers that this is code is intentionally hacky, so as to work around the described behavior.